pkg/math/oracle/median.go

```diff
@@ -97,80 +102,76 @@ func (m *MedianAggregator) AggregateData() {
 
 		// Take the median of the converted prices. This takes the average of the middle two
 		// prices if the number of prices is even.
-		price := median.CalculateMedian(convertedPrices)
-		updatedPrices[target] = price
+		price := math.CalculateMedian(convertedPrices)
+		indexPrices[target.String()] = new(big.Float).Copy(price)
+
+		// Scale the price to the target ticker's decimals.
+		scaledPrices[target.String()] = math.ScaleBigFloat(new(big.Float).Copy(price), target.Decimals)
+
 		m.logger.Info(
 			"calculated median price",
-			zap.String("ticker", ticker),
-			zap.String("price", price.String()),
+			zap.String("target_ticker", ticker),
+
+			zap.String("unscaled_price", indexPrices[target.String()].String()),
+			zap.String("scaled_price", scaledPrices[target.String()].String()),
 			zap.Any("converted_prices", convertedPrices),
 		)
-		m.metrics.AddTickerTick(target.String())
-
 		floatPrice, _ := price.Float64()
+		m.metrics.AddTickerTick(target.String())
 		m.metrics.UpdateAggregatePrice(target.String(), target.GetDecimals(), floatPrice)
 	}
 
 	// Update the aggregated data. These prices are going to be used as the index prices the
 	// next time we calculate prices.
-	m.DataAggregator.SetAggregatedData(updatedPrices)
-	m.logger.Info("calculated median prices for price feeds", zap.Int("num_prices", len(updatedPrices)))
+	m.logger.Info("calculated median prices for price feeds", zap.Int("num_prices", len(indexPrices)))
+	m.indexPrices = indexPrices
+	m.scaledPrices = scaledPrices
 }
 
 // CalculateConvertedPrices calculates the converted prices for a given set of paths and target ticker.
 // The prices utilized are the prices most recently seen by the providers. Each price is within a
 // MaxPriceAge window so is safe to use.
 func (m *MedianAggregator) CalculateConvertedPrices(
 	target mmtypes.Ticker,
-	paths mmtypes.Paths,
-) []*big.Int {
+	providers []mmtypes.ProviderConfig,
```

Comment:
can this function just take in `mmtypes.Market`?

which is just the composition of these two types?